server/services/PostsService.js

```diff
@@ -241,69 +251,69 @@ const getPostsByQuery = async (query, paginateId, postsLimit) => {
   }
 }
 
-const updateLikes = function (postId, userId) {
+const updateLikes = async (postId, userId) => {
   const postRef = db.collection('posts').doc(postId)
+  return await db.runTransaction(async t => {
+    const docPost = await t.get(postRef)
+    const postOwnerId = docPost.data().userId
+    const userPostsRef = db.collection('users-posts').doc(postOwnerId)
+    const myPostsRef = db.collection('users-posts').doc(userId)
+    const userPostsDoc = await t.get(userPostsRef)
+    const myPostsDoc = await t.get(myPostsRef)
+    const {
+      likes,
+      likesCount
+    } = docPost.data()
+    const {
+      posts
+    } = userPostsDoc.data()
+    const {
+      likedPosts
+    } = myPostsDoc.data()
+    let like = (!likedPosts.includes(postId)) ? true : false
+    const newLike = {
+      userId: userId,
+      date: Date.now()
+    }
 
-  return db
-    .runTransaction(t => {
-      return t.get(postRef).then(docPost => {
-        const postOwnerId = docPost.data().userId
-        const userPostsRef = db.collection('users-posts').doc(postOwnerId)
-        return t.get(userPostsRef).then(userPostsDoc => {
-          const {
-            likes,
-            likesCount
-          } = docPost.data()
-          const {
-            posts,
-            likedPosts
-          } = userPostsDoc.data().posts
-
-          let like = (likedPosts.includes(postId) && likes.includes(userId))? false : true
-
-          const newLikes = (like) ? likes.push(userId) : likes.filter(like => like !== userId)
-          const newLikesCount = (like) ? likesCount++ : likesCount--
-          const newLikedPosts = (like) ? likedPosts.push(postId) : likedPosts.filter(like => like !== postId)
-          const newPosts = posts.map(post => {
-            if (post.postId === postId) {
-              (like) ? post.likes++ : post.likes--
-            }
-            return post
-          })
-          
-          t.update(postRef, {
-            likes: newLikes,
-            likesCount: newLikesCount
-          })
-          t.update(userPostsRef, {
-            likedPosts: newLikedPosts,
-            posts: newPosts
-          })
-
-          return like ? 'like' : 'dislike'
-        })
-      })
+    const newLikes = (like) ? [...likes, newLike] : likes.filter(el => el.userId !== userId)
+    const newLikesCount = (like) ? likesCount + 1 : likesCount - 1
+    const newLikedPosts = (like) ? [...likedPosts, postId] : likedPosts.filter(el => el !== postId)
```

Comment:
Usually something that is ends as '*Posts' inclines having an array of objects, in your case this is an array of strings.
For future use try renaming things like this into '*PostIds'